File: src/CSPAF/CSPAgents/CSPUtilityServerGroups/CSPDownloadAgentBootstrap.cs
using System;
using System.Collections.Generic;

using CloudSuitePortal;
using CloudSuitePortal.Models;
using CloudSuitePortal.Services;
using CloudSuitePortal.Controllers;

namespace actions.CSP.CSPAgents.CSPUtilityServerGroups
{
    class DownloadAgentBootstrap
    {
        public void run(Dictionary<string, object> Params)
        {
            var navigationController = CSPBrowser.Application.WindowController.NavigationController;
            var controller = (CSPUtilityServerGroupsController)navigationController.VisibleViewController;
            controller.DownloadAgentBootstrap();
        }
    }
}

File: src/CSPAF/CSPAgents/CSPUtilityServerGroups/CSPDownloadAgentBootstrap.java
package CSPAF.CSPAgents.CSPUtilityServerGroups;

import java.io.*;
import java.util.*;

import com.infor.cloudsuite.portal.core.views.*;
import com.infor.cloudsuite.portal.core.services.*;
import com.infor.cloudsuite.portal.core.controllers.*;
import com.infor.cloudsuite.portal.singletenant.agents.controllers.*;

class CSPDownloadAgentBootstrap
{
    public void run(HashMap<String, Object> params) throws IOException, CSPTable.RowNotFoundException
    {
        CSPNavigationController navigationController = CSPBrowser.getApplication().getWindowController().getNavigationController();
        CSPUtilityServerGroupsController controller = (CSPUtilityServerGroupsController)navigationController.getVisibleViewController();
        controller.downloadAgentBootstrap();
    }
}